### ISSUE TITLE
Do not check autogenerated code of WF activities

### ISSUE THREAD

**Issue by knoxi** [DotNetAnalyzers/StyleCopAnalyzers#1479]
We have a project which contains workflow activities which are created with the designer.

Now the analyzer checks the autogenerated code files like following example for an activity:
.\obj\Debug\namepsace_activity_BeforeInitializeComponentHelper.cs
.\obj\Debug\namespace_activity_CompiledExpressionRoot.cs

Please ignore this autogenerated files in the analysis.

**Comment by sharwell** [comment]
Hi @knoxi,

I'm trying to reproduce this issue but so far have not been successful. I used the following steps:

1. Create a new **Activity Library** project (targeted .NET Framework 4.6).
2. Added a Messaging &rarr; Receive block to the default **Activity1.xaml** file.
3. Built the project.

I saw two files were created with names like you describe. However, both of these files were already excluded from analysis:

1. **obj\Debug\ActivityLibrary1_Activity1_BeforeInitializeComponentHelper.cs**: excluded for having `<auto-generated>` in the file header.
2. **obj\Debug\ActivityLibrary1_Activity1_CompiledExpressionRoot.cs**: excluded because it was an empty file.

Can you provide some additional information to help us track this down:

1. Do your steps to reproduce this differ from the ones I used?
2. What is one specific warning you are seeing reported for these files (most of our analyzers are implemented in their own classes so one might have been missed)?

Thank you for taking the time to provide feedback. It really helps us make the product better! :smile:

**Comment by knoxi** [comment]
You're right, I didn't see apperaring the <b>_BeforeInitializeComponentHelper.cs</b> in my error list

I think you should add some InArguments / OutArguments to the activity and use variables in some steps, then the <b>_CompiledExpressionRoot.cs</b> file isn't empty.

**Comment by knoxi** [comment]
Here the code of my sample activity:

<Activity mc:Ignorable="sap sap2010 sads" x:Class="WF_Activity_SA.Activities.Activity1"
 xmlns="http://schemas.microsoft.com/netfx/2009/xaml/activities"
 xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
 xmlns:mca="clr-namespace:Microsoft.CSharp.Activities;assembly=System.Activities"
 xmlns:sads="http://schemas.microsoft.com/netfx/2010/xaml/activities/debugger"
 xmlns:sap="http://schemas.microsoft.com/netfx/2009/xaml/activities/presentation"
 xmlns:sap2010="http://schemas.microsoft.com/netfx/2010/xaml/activities/presentation"
 xmlns:scg="clr-namespace:System.Collections.Generic;assembly=mscorlib"
 xmlns:sco="clr-namespace:System.Collections.ObjectModel;assembly=mscorlib"
 xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
  <x:Members>
    <x:Property Name="RunningCount" Type="InArgument(x:Int32)">
      <x:Property.Attributes>
        <RequiredArgumentAttribute />
      </x:Property.Attributes>
    </x:Property>
  </x:Members>
  <sap2010:ExpressionActivityEditor.ExpressionActivityEditor>C#</sap2010:ExpressionActivityEditor.ExpressionActivityEditor>
  <sap2010:WorkflowViewState.IdRef>WF_Activity_SA.Activities.Activity1_1</sap2010:WorkflowViewState.IdRef>
  <TextExpression.NamespacesForImplementation>
    <sco:Collection x:TypeArguments="x:String">
      <x:String>System</x:String>
      <x:String>System.Collections.Generic</x:String>
      <x:String>System.Data</x:String>
      <x:String>System.Linq</x:String>
      <x:String>System.Text</x:String>
    </sco:Collection>
  </TextExpression.NamespacesForImplementation>
  <TextExpression.ReferencesForImplementation>
    <sco:Collection x:TypeArguments="AssemblyReference">
      <AssemblyReference>System</AssemblyReference>
      <AssemblyReference>System.Core</AssemblyReference>
      <AssemblyReference>System.Xml.Linq</AssemblyReference>
      <AssemblyReference>System.Data.DataSetExtensions</AssemblyReference>
      <AssemblyReference>Microsoft.CSharp</AssemblyReference>
      <AssemblyReference>System.Data</AssemblyReference>
      <AssemblyReference>System.Net.Http</AssemblyReference>
      <AssemblyReference>System.Xml</AssemblyReference>
      <AssemblyReference>System.Activities</AssemblyReference>
      <AssemblyReference>System.ServiceModel</AssemblyReference>
      <AssemblyReference>System.ServiceModel.Activities</AssemblyReference>
      <AssemblyReference>System.Xaml</AssemblyReference>
      <AssemblyReference>mscorlib</AssemblyReference>
      <AssemblyReference>WF_Activity_SA</AssemblyReference>
    </sco:Collection>
  </TextExpression.ReferencesForImplementation>
  <Sequence DisplayName="Sequence 1" sap2010:WorkflowViewState.IdRef="Sequence_1">
    <Sequence.Variables>
      <Variable x:TypeArguments="x:Boolean" Name="isRunning" />
    </Sequence.Variables>
    <DoWhile sap2010:WorkflowViewState.IdRef="DoWhile_1">
      <DoWhile.Variables>
        <Variable x:TypeArguments="x:Int32" Name="iteration" />
      </DoWhile.Variables>
      <DoWhile.Condition>
        <mca:CSharpValue x:TypeArguments="x:Boolean">iteration &lt;= RunningCount</mca:CSharpValue>
      </DoWhile.Condition>
      <Sequence sap2010:WorkflowViewState.IdRef="Sequence_2">
        <Delay Duration="50.00:00:00" sap2010:WorkflowViewState.IdRef="Delay_1" />
        <Assign sap2010:WorkflowViewState.IdRef="Assign_1">
          <Assign.To>
            <OutArgument x:TypeArguments="x:Int32">
              <mca:CSharpReference x:TypeArguments="x:Int32">iteration</mca:CSharpReference>
            </OutArgument>
          </Assign.To>
          <Assign.Value>
            <InArgument x:TypeArguments="x:Int32">
              <mca:CSharpValue x:TypeArguments="x:Int32">iteration + 1;</mca:CSharpValue>
            </InArgument>
          </Assign.Value>
        </Assign>
      </Sequence>
    </DoWhile>
    <sads:DebugSymbol.Symbol>d0hDOlxQcm9qZWN0c1xUZXN0XFdGX0FjdGl2aXR5X1NBXFdGX0FjdGl2aXR5X1NBXEFjdGl2aXRpZXNcQWN0aXZpdHkxLnhhbWwJLwNLDgIBATMFSQ8CAQI4CThkAgEPOgdIEgIBAzsJO1MCAQ08CUcSAgEEOxk7JgIBDkQPRFoCAQk/Dz9dAgEF</sads:DebugSymbol.Symbol>
  </Sequence>
  <sap2010:WorkflowViewState.ViewStateManager>
    <sap2010:ViewStateManager>
      <sap2010:ViewStateData Id="Delay_1" sap:VirtualizedContainerService.HintSize="242,22" />
      <sap2010:ViewStateData Id="Assign_1" sap:VirtualizedContainerService.HintSize="242,62" />
      <sap2010:ViewStateData Id="Sequence_2" sap:VirtualizedContainerService.HintSize="264,248">
        <sap:WorkflowViewStateService.ViewState>
          <scg:Dictionary x:TypeArguments="x:String, x:Object">
            <x:Boolean x:Key="IsExpanded">True</x:Boolean>
          </scg:Dictionary>
        </sap:WorkflowViewStateService.ViewState>
      </sap2010:ViewStateData>
      <sap2010:ViewStateData Id="DoWhile_1" sap:VirtualizedContainerService.HintSize="464,408" />
      <sap2010:ViewStateData Id="Sequence_1" sap:VirtualizedContainerService.HintSize="486,532">
        <sap:WorkflowViewStateService.ViewState>
          <scg:Dictionary x:TypeArguments="x:String, x:Object">
            <x:Boolean x:Key="IsExpanded">True</x:Boolean>
          </scg:Dictionary>
        </sap:WorkflowViewStateService.ViewState>
      </sap2010:ViewStateData>
      <sap2010:ViewStateData Id="WF_Activity_SA.Activities.Activity1_1" sap:VirtualizedContainerService.HintSize="526,612" />
    </sap2010:ViewStateManager>
  </sap2010:WorkflowViewState.ViewStateManager>
</Activity>

**Comment by sharwell** [comment]
> How can I share the sample activity?

I was in the middle of editing your comment to fix the formatting. If you paste it again I'll make it look right.

**Comment by knoxi** [comment]
the activity code:

<Activity mc:Ignorable="sap sap2010 sads" x:Class="WF_Activity_SA.Activities.Activity1"
 xmlns="http://schemas.microsoft.com/netfx/2009/xaml/activities"
 xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
 xmlns:mca="clr-namespace:Microsoft.CSharp.Activities;assembly=System.Activities"
 xmlns:sads="http://schemas.microsoft.com/netfx/2010/xaml/activities/debugger"
 xmlns:sap="http://schemas.microsoft.com/netfx/2009/xaml/activities/presentation"
 xmlns:sap2010="http://schemas.microsoft.com/netfx/2010/xaml/activities/presentation"
 xmlns:scg="clr-namespace:System.Collections.Generic;assembly=mscorlib"
 xmlns:sco="clr-namespace:System.Collections.ObjectModel;assembly=mscorlib"
 xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
  <x:Members>
    <x:Property Name="RunningCount" Type="InArgument(x:Int32)">
      <x:Property.Attributes>
        <RequiredArgumentAttribute />
      </x:Property.Attributes>
    </x:Property>
  </x:Members>
  <sap2010:ExpressionActivityEditor.ExpressionActivityEditor>C#</sap2010:ExpressionActivityEditor.ExpressionActivityEditor>
  <sap2010:WorkflowViewState.IdRef>WF_Activity_SA.Activities.Activity1_1</sap2010:WorkflowViewState.IdRef>
  <TextExpression.NamespacesForImplementation>
    <sco:Collection x:TypeArguments="x:String">
      <x:String>System</x:String>
      <x:String>System.Collections.Generic</x:String>
      <x:String>System.Data</x:String>
      <x:String>System.Linq</x:String>
      <x:String>System.Text</x:String>
    </sco:Collection>
  </TextExpression.NamespacesForImplementation>
  <TextExpression.ReferencesForImplementation>
    <sco:Collection x:TypeArguments="AssemblyReference">
      <AssemblyReference>System</AssemblyReference>
      <AssemblyReference>System.Core</AssemblyReference>
      <AssemblyReference>System.Xml.Linq</AssemblyReference>
      <AssemblyReference>System.Data.DataSetExtensions</AssemblyReference>
      <AssemblyReference>Microsoft.CSharp</AssemblyReference>
      <AssemblyReference>System.Data</AssemblyReference>
      <AssemblyReference>System.Net.Http</AssemblyReference>
      <AssemblyReference>System.Xml</AssemblyReference>
      <AssemblyReference>System.Activities</AssemblyReference>
      <AssemblyReference>System.ServiceModel</AssemblyReference>
      <AssemblyReference>System.ServiceModel.Activities</AssemblyReference>
      <AssemblyReference>System.Xaml</AssemblyReference>
      <AssemblyReference>mscorlib</AssemblyReference>
      <AssemblyReference>WF_Activity_SA</AssemblyReference>
    </sco:Collection>
  </TextExpression.ReferencesForImplementation>
  <Sequence DisplayName="Sequence 1" sap2010:WorkflowViewState.IdRef="Sequence_1">
    <Sequence.Variables>
      <Variable x:TypeArguments="x:Boolean" Name="isRunning" />
    </Sequence.Variables>
    <DoWhile sap2010:WorkflowViewState.IdRef="DoWhile_1">
      <DoWhile.Variables>
        <Variable x:TypeArguments="x:Int32" Name="iteration" />
      </DoWhile.Variables>
      <DoWhile.Condition>
        <mca:CSharpValue x:TypeArguments="x:Boolean">iteration &lt;= RunningCount</mca:CSharpValue>
      </DoWhile.Condition>
      <Sequence sap2010:WorkflowViewState.IdRef="Sequence_2">
        <Delay Duration="50.00:00:00" sap2010:WorkflowViewState.IdRef="Delay_1" />
        <Assign sap2010:WorkflowViewState.IdRef="Assign_1">
          <Assign.To>
            <OutArgument x:TypeArguments="x:Int32">
              <mca:CSharpReference x:TypeArguments="x:Int32">iteration</mca:CSharpReference>
            </OutArgument>
          </Assign.To>
          <Assign.Value>
            <InArgument x:TypeArguments="x:Int32">
              <mca:CSharpValue x:TypeArguments="x:Int32">iteration + 1;</mca:CSharpValue>
            </InArgument>
          </Assign.Value>
        </Assign>
      </Sequence>
    </DoWhile>
    <sads:DebugSymbol.Symbol>d0hDOlxQcm9qZWN0c1xUZXN0XFdGX0FjdGl2aXR5X1NBXFdGX0FjdGl2aXR5X1NBXEFjdGl2aXRpZXNcQWN0aXZpdHkxLnhhbWwJLwNLDgIBATMFSQ8CAQI4CThkAgEPOgdIEgIBAzsJO1MCAQ08CUcSAgEEOxk7JgIBDkQPRFoCAQk/Dz9dAgEF</sads:DebugSymbol.Symbol>
  </Sequence>
  <sap2010:WorkflowViewState.ViewStateManager>
    <sap2010:ViewStateManager>
      <sap2010:ViewStateData Id="Delay_1" sap:VirtualizedContainerService.HintSize="242,22" />
      <sap2010:ViewStateData Id="Assign_1" sap:VirtualizedContainerService.HintSize="242,62" />
      <sap2010:ViewStateData Id="Sequence_2" sap:VirtualizedContainerService.HintSize="264,248">
        <sap:WorkflowViewStateService.ViewState>
          <scg:Dictionary x:TypeArguments="x:String, x:Object">
            <x:Boolean x:Key="IsExpanded">True</x:Boolean>
          </scg:Dictionary>
        </sap:WorkflowViewStateService.ViewState>
      </sap2010:ViewStateData>
      <sap2010:ViewStateData Id="DoWhile_1" sap:VirtualizedContainerService.HintSize="464,408" />
      <sap2010:ViewStateData Id="Sequence_1" sap:VirtualizedContainerService.HintSize="486,532">
        <sap:WorkflowViewStateService.ViewState>
          <scg:Dictionary x:TypeArguments="x:String, x:Object">
            <x:Boolean x:Key="IsExpanded">True</x:Boolean>
          </scg:Dictionary>
        </sap:WorkflowViewStateService.ViewState>
      </sap2010:ViewStateData>
      <sap2010:ViewStateData Id="WF_Activity_SA.Activities.Activity1_1" sap:VirtualizedContainerService.HintSize="526,612" />
    </sap2010:ViewStateManager>
  </sap2010:WorkflowViewState.ViewStateManager>
</Activity>

**Comment by sharwell** [comment]
Thanks, I was able to reproduce the issue.

This appears to be a bug in a build extension associated with Workflow. As you can see from all the other files generated from XAML, the expectation was to add an `<auto-generated>` header to each of them. I think the best way to approach this is to add the following to the end of your **.csproj** file, right before `</Project>`:

```xml
<!--
  Add an auto-generated header to files generated by XAML extensions. Various code analysis tools rely on this
  header to identify generated files.
-->
<Target Name="AddAutoGeneratedComment" AfterTargets="XamlMarkupCompilePass2"
        Condition="'@(ExtensionGeneratedCodeFiles)' != ''">
  <Exec Command="for %%f in (@(ExtensionGeneratedCodeFiles)) do echo // ^&lt;auto-generated^&gt; > %%f.temp" />
  <Exec Command="for %%f in (@(ExtensionGeneratedCodeFiles)) do type %%f >> %%f.temp" />
  <Exec Command="for %%f in (@(ExtensionGeneratedCodeFiles)) do move /y %%f.temp %%f" />
  <Message Text="XAML Generated Code Warnings Removed: @(ExtensionGeneratedCodeFiles)" />
</Target>
```

**Comment by sharwell** [comment]
I'm going to close this as a duplicate of #1430, just so we can keep discussions about detecting auto-generated code limited to one thread.
